core/src/main/java/org/apache/calcite/rel/AbstractRelNode.java

```diff
@@ -363,11 +352,14 @@ public void replaceInput(
   }
 
   public String toString() {
-    return desc;
+    return getDescription();
   }
 
+  /* Description, consists of id plus digest */
   public final String getDescription() {
-    return desc;
+    StringBuilder sb = new StringBuilder("rel#");
+    sb = sb.append(id).append(':').append(digest);
+    return sb.toString();
```

Comment:
If this is called before recomputeDigest(), we will get a different description string than the original implementation.  Is that you want?

Reply:
Yes. That is expected.

Reply:
I'm wondering if we can add a benchmark test for the toString with/without the fix.